docs/stream.md

```diff
@@ -120,13 +120,13 @@ Argument     | Type                 | Required | Description
 
 Creates a new stream with the results of calling the function on every value in the stream with an accumulator and the incoming value.
 
-Note that you can prevent dependent streams from being updated by returning the special value `stream.HALT` inside the accumulator function.
+Note that you can prevent dependent streams from being updated by returning the special value `stream.SKIP` inside the accumulator function.
 
 `stream = Stream.scan(fn, accumulator, stream)`
 
 Argument      | Type                             | Required | Description
 ------------- | -------------------------------- | -------- | ---
-`fn`          | `(accumulator, value) -> result \| HALT` | Yes      | A function that takes an accumulator and value parameter and returns a new accumulator value
+`fn`          | `(accumulator, value) -> result \| SKIP` | Yes      | A function that takes an accumulator and value parameter and returns a new accumulator value
```

Comment:
Technically the type should be

```
scan :: ((a, b) -> a) -> a -> Stream b -> Stream a
```

If you don't return a value of the same type as the first argument in the accumulator function
you're going to get into trouble.

Reply:
Do you think it would be enough to replace `result` with `value`, and add ` of the same type` to the description?

Reply:
It would be nice to add type signatures to the docs generally like in your example above, but perhaps that should be another pull request?

Reply:
I think `result` is okay if the prose explains it's also plugged back into `accumulator`.

Reply:
Yeah, I think prose is the correct way to handle this for now

Reply:
@porsager If you want to suggest changing the type signatures, an issue would be a better place to suggest and explain that.